tests/returntypes/test_parameter_shift_shot_vec_new.py

```diff
@@ -1333,18 +1361,20 @@ def test_involutory_variance_multi_param(self, tol):
 
             assert len(tapes) == 1 + 2 * 2
 
-            # TODO: finite diff
-            # tapes, fn = qml.gradients.finite_diff(tape)
-            # gradF = fn(dev.batch_execute(tapes))
-            # assert len(tapes) == 3
+        tapes, fn = qml.gradients.finite_diff(tape, h=10e-2, shots=shot_vec)
+        all_res = fn(dev.batch_execute(tapes))
+        for gradF in all_res:
+
+            assert len(tapes) == 3
 
             expected = 2 * np.sin(a + b) * np.cos(a + b)
-            # assert gradF[0] == pytest.approx(expected, abs=shot_vec_tol)
-            assert gradA[0] == pytest.approx(expected, abs=shot_vec_tol)
+            assert gradF[0] == pytest.approx(expected, abs=finite_diff_tol)
+            assert gradA[0] == pytest.approx(expected, abs=finite_diff_tol)
 
-            # assert gradF[1] == pytest.approx(expected, abs=shot_vec_tol)
-            assert gradA[1] == pytest.approx(expected, abs=shot_vec_tol)
+            assert gradF[1] == pytest.approx(expected, abs=finite_diff_tol)
+            assert gradA[1] == pytest.approx(expected, abs=finite_diff_tol)
 
+    @flaky(max_runs=5)
```

Comment:
Interesting that this needs to be marked flaky when the finite difference tests were added 🤔. Since the parameter-shift tests use the same shot vector, I'd guess that this is due to the step size `h=0.1`? What happens if we decrease it?

Reply:
If `h` is decreased, the values will be extremely off (on the order of +/- thousands). As a matter of fact, the default makes the test results to be completely off, hence the specific `h` values used. This was the best compromise I could find.

Note, I'm not sure we had too many finite differences test cases with finite shots, let alone shot vectors in the code base.